app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -474,8 +484,47 @@ class SectionRenderingService(
         htmlForFileUpload(fieldValue, formTemplateId, index, ei, data, userDetails, maybeValidated, lang)
       case InformationMessage(infoType, infoText) =>
         htmlForInformationMessage(fieldValue, infoType, infoText, index, ei)
+      case HmrcTaxPeriod(a, b, c) =>
+        htmlForHmrcTaxPeriod(fieldValue, index, ei, maybeValidated, data, obligations, a, b, c)
     }
 
+  private def htmlForHmrcTaxPeriod(
+    fieldValue: FormComponent,
+    index: Int,
+    ei: ExtraInfo,
+    validatedType: ValidatedType,
+    data: FormDataRecalculated,
+    obligations: Map[HmrcTaxPeriodIdentifier, TaxPeriods],
+    idType: String,
+    idNumber: String,
+    regimeType: String) = {
```

Comment:
Why we passing `idType`, `idNumber` and `regimeType` separately? To pass `HmrcTaxPeriodIdentifier` seems to be better option.